rig-core/examples/multi_turn_agent.rs

```diff
@@ -0,0 +1,285 @@
+use rig::{
+    agent::Agent,
+    completion::{self, Completion, PromptError, ToolDefinition},
+    message::{AssistantContent, Message, ToolCall, ToolFunction, ToolResultContent, UserContent},
+    providers::anthropic,
+    tool::Tool,
+    OneOrMany,
+};
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+
+struct MultiTurnAgent<M: rig::completion::CompletionModel> {
+    agent: Agent<M>,
+    chat_history: Vec<completion::Message>,
+}
+
+impl<M: rig::completion::CompletionModel> MultiTurnAgent<M> {
+    async fn multi_turn_prompt(
+        &mut self,
+        prompt: impl Into<Message> + Send,
+    ) -> Result<String, PromptError> {
+        let mut current_prompt: Message = prompt.into();
+        loop {
+            println!("Current Prompt: {:?}\n", current_prompt);
+            let resp = self
+                .agent
+                .completion(current_prompt.clone(), self.chat_history.clone())
+                .await?
+                .send()
+                .await?;
+
+            let mut final_text = None;
+
+            for choice in resp.choice.into_iter() {
+                match choice {
+                    AssistantContent::Text(text) => {
+                        println!("Intermediate Response: {:?}\n", text.text);
+                        final_text = Some(text.text.clone());
+                        self.chat_history.push(current_prompt.clone());
+                        let response_message = Message::Assistant {
+                            content: OneOrMany::one(AssistantContent::text(&text.text)),
+                        };
+                        self.chat_history.push(response_message);
+                    }
+                    AssistantContent::ToolCall(ToolCall {
+                        id,
+                        function: ToolFunction { name, arguments },
+                    }) => {
+                        self.chat_history.push(current_prompt.clone());
+                        let tool_call_msg = Message::Assistant {
+                            content: OneOrMany::one(AssistantContent::tool_call(
+                                &id,
+                                &name,
+                                arguments.clone(),
+                            )),
+                        };
```

Comment:
Could you just reuse the `AssistantContent::ToolCall(content)` from the match pattern when defining `tool_call_msg`?

Reply:
I don't follow, what's the suggestions here? to add like a `Message::tool_call(...)`? or something else?

Reply:
```rust
AssistantContent::ToolCall(content) => {
    self.chat_history.push(current_prompt.clone());
    let tool_call_msg = Message::Assistant {content};
    ...
}
```